File: CS-Examples/22_TextBoxes/ReplaceTextInTextBox.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;
using Spire.Xls.Core;

namespace ReplaceTextInTextBox
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Load the Excel document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ReplaceTextInTextBox.xlsx");

            // Get the first worksheet from the workbook
            Worksheet sheet = workbook.Worksheets[0];

            string tag = "TAG_1$TAG_2";
            string replace = "Spire.XLS for .NET$Spire.XLS for JAVA";

            for (int i = 0; i < tag.Split('$').Length; i++)
            {
                // Replace text in textbox
                ReplaceTextInTextBox(sheet, "<" + tag.Split('$')[i] + ">", replace.Split('$')[i]);
            }

            // Specify the output file path
            string output = "ReplaceTextInTextBox_out.xlsx";

            // Save the modified workbook to a file
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the Excel file
            ExcelDocViewer(output);
        }
        private void ReplaceTextInTextBox(Worksheet sheet, string sFind, string sReplace)
        {
            for (int i = 0; i < sheet.TextBoxes.Count; i++)
            {
                ITextBox tb = sheet.TextBoxes[i];
                if (!String.IsNullOrEmpty(tb.Text))
                {
                    if (tb.Text.Contains(sFind))
                    {
                        tb.Text = tb.Text.Replace(sFind, sReplace);
                    }
                }
            }
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

    }
}

File: CS-Examples/22_TextBoxes/ReplaceTextInTextBox.vb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls
Imports Spire.Xls.Core

Namespace ReplaceTextInTextBox
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Load the workbook from a file
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ReplaceTextInTextBox.xlsx")

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Define the tag and replace strings
            Dim tag As String = "TAG_1$TAG_2"
            Dim replace As String = "Spire.XLS for .NET$Spire.XLS for JAVA"

            ' Iterate through each tag and replace its corresponding value in the worksheet
            For i As Integer = 0 To tag.Split("$"c).Length - 1
                ReplaceTextInTextBox(sheet, "<" & tag.Split("$"c)(i) & ">", replace.Split("$"c)(i))
            Next i

            ' Specify the output filename
            Dim output As String = "ReplaceTextInTextBox_out.xlsx"

            ' Save the modified workbook to a file in Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(output)
		End Sub
		Private Sub ReplaceTextInTextBox(ByVal sheet As Worksheet, ByVal sFind As String, ByVal sReplace As String)
			For i As Integer = 0 To sheet.TextBoxes.Count - 1
				Dim tb As ITextBox = sheet.TextBoxes(i)
				If Not String.IsNullOrEmpty(tb.Text) Then
					If tb.Text.Contains(sFind) Then
						tb.Text = tb.Text.Replace(sFind, sReplace)
					End If
				End If
			Next i
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
